expyfun/stimuli/_tracker.py

```diff
@@ -259,18 +268,25 @@ def respond(self, correct):
                     x=[int(s) for s in self._x])))
 
     def _stop_here(self):
-        if self._stop_rule.lower() == 'reversals':
-            self._n_stop = self._n_reversals
-        elif self._stop_rule.lower() == 'trials':
-            self._n_stop = self._n_trials
-        return self._n_stop >= self._stop_criterion
+        if self._n_reversals > self._stop_reversals:
+            self._n_stop = True
+        elif self._n_trials > self._stop_trials:
+            self._n_stop = True
+        else:
+            self._n_stop = False
+        return self._n_stop
 
     def _step_index(self):
         if self._change_rule.lower() == 'reversals':
             self._n_change = self._n_reversals
-        elif self._stop_rule.lower() == 'trials':
+        elif self._change_rule.lower() == 'trials':
             self._n_change = self._n_trials
-        return np.where(self._n_change >= self._change_criteria)[0][-1]
+        step_index = np.where(self._n_change >= self._change_indices)[0]
+        if len(step_index) == 0 or self._change_indices == [0]:
```

Comment:
I get an error on this line:
```
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
This is because `self._change_indices == np.array([4, 8])`. Please fix and add a test that would have exposed this error.